video/out/gpu/video.c

```diff
@@ -1338,18 +1338,20 @@ static const char *get_tex_swizzle(struct image *img)
 
 // Copy a texture to the vec4 color, while increasing offset. Also applies
 // the texture multiplier to the sampled color
-static void copy_image(struct gl_video *p, int *offset, struct image img)
+static void copy_image(struct gl_video *p, unsigned int *offset, struct image img)
 {
-    int count = img.components;
-    assert(*offset + count <= 4);
-    assert(img.padding + count <= 4);
-
-    int id = pass_bind(p, img);
+    const unsigned int count = img.components;
     char src[5] = {0};
     char dst[5] = {0};
+
+    assert(*offset + count < sizeof(dst));
+    assert(img.padding + count < sizeof(src));
+
+    int id = pass_bind(p, img);
+
     const char *tex_fmt = get_tex_swizzle(&img);
     const char *dst_fmt = "rgba";
-    for (int i = 0; i < count; i++) {
+    for (unsigned int i = 0; i < count; i++) {
```

Comment:
The compiler knows `i >= 0` here and shouldn't produce a warning.

Reply:
../video/out/gpu/video.c: In function ‘copy_image’:
../video/out/gpu/video.c:1354:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
 1354 |         dst[i] = dst_fmt[*offset + i];
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
../video/out/gpu/video.c:1349:10: note: at offset 5 into destination object ‘dst’ of size 5
 1349 |     char dst[5] = {0};
      |          ^~~
../video/out/gpu/video.c:1354:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
 1354 |         dst[i] = dst_fmt[*offset + i];
      |         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
../video/out/gpu/video.c:1349:10: note: at offset 6 into destination object ‘dst’ of size 5
 1349 |     char dst[5] = {0};
      |          ^~~